### ISSUE TITLE
Phenotype Phase Plane plot_matplotlib does not work in Python 3

### ISSUE THREAD

**Issue by cdiener** [opencobra/cobrapy#201]
The example from phenotype_phase_plane.ipynb will fail in Python 3 in `plot_matplotlib` with a ValueError in the `colors` array. The problem here is the initialization of the array:

```Python
colors = empty(self.growth_rates.shape, dtype="|S7")
```

This casts to a normal string in Python 2 but casts to a byte-string in Python 3 which is not supported by matplotlib in Python 3 (expects unicode). This can be fixed by deriving the correct numpy dtype at runtime using:

```Python
from numpy import ..., dtype
....
colors = empty(self.growth_rates.shape, dtype=dtype((str, 7)))
```

A probably bigger problem is the absence of tests for the entire module...

Should I create a pull request?

Cheers
Christian

**Comment by aebrahim** [comment]
Yes please create a pull request. That would be very helpful.

There actually are tests for the calculation part phenotypte phase planes, but the plotting isn't tested because that requires setting up matplotlib on travis. Figuring that out would be helpful as well.

**Comment by cdiener** [comment]
Hey, and here I am again complaining about things that aren't true :) Okay I will make a pull for the fix and experiment with the tests...

Yeah matplotlib is tricky to compile. Isn there a python(3)-matplotlib package like in Debian? Otherwise I think that gets resolved with #195 since you could install it with conda...

**Comment by aebrahim** [comment]
Closed in  2565431
